jdbc/jooq/src/main/java/example/springdata/jdbc/jooq/CategoryConfiguration.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package example.springdata.jdbc.jooq;
+
+import org.jooq.impl.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.jdbc.repository.config.EnableJdbcRepositories;
+import org.springframework.data.jdbc.repository.config.JdbcConfiguration;
+import org.springframework.data.relational.core.mapping.event.BeforeSaveEvent;
+import org.springframework.data.relational.core.mapping.event.RelationalEvent;
+import org.springframework.jdbc.datasource.TransactionAwareDataSourceProxy;
+
+import javax.sql.DataSource;
+
+/**
+ * Contains infrastructure necessary for creating repositories and two listeners.
+ * <p>
+ * Not that a listener may change an entity without any problem.
+ *
+ * @author Jens Schauder
+ * @author Mark Paluch
+ * @author Florian Lüdiger
+ */
+@Configuration
+@EnableJdbcRepositories
+@Import(JdbcConfiguration.class)
+public class CategoryConfiguration {
+
+	@Autowired
+	private DataSource dataSource;
+
+	@Bean
+	public ApplicationListener<?> loggingListener() {
+
+		return (ApplicationListener<ApplicationEvent>) event -> {
+			if (event instanceof RelationalEvent) {
+				System.out.println("Received an event: " + event);
+			}
+		};
+	}
+
+	@Bean
+	public ApplicationListener<BeforeSaveEvent> timeStampingSaveTime() {
```

Comment:
Please remove all code not essential for this example. Simplify the model if necessary.

Reply:
done